docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md

```diff
@@ -106,3 +106,34 @@ https://github.com/dotnet/roslyn/issues/57750
     void M(IEnumerable<char> s = "hello")
     ```
 
+8. <a name="8"></a>Starting with Visual Studio 17.2, an async `foreach` prefers to bind using a pattern-based `DisposeAsync()` method rather than `IAsyncDisposable.DisposeAsync()`.
+    For instance, the `DisposeAsync()` will be picked, rather than the `IAsyncEnumerator<int>.DisposeAsync()` method on `AsyncEnumerator`:
+    ```csharp
+    await foreach (var i in new AsyncEnumerable())
+    {
+    }
+
+    struct AsyncEnumerable
```

Comment:
Is `AsyncEnumerable` required to implement `IAsyncEnumerable<int>`?

Reply:
No. This sample illustrate pattern-based enumeration and a choice for method of disposal (pattern-based or interface-based).